rosidl_typesupport_cpp/resource/srv__type_support.cpp.em

```diff
@@ -109,12 +117,86 @@ static const type_support_map_t _@(service.namespaced_type.name)_service_typesup
   &_@(service.namespaced_type.name)_service_typesupport_data.data[0],
 };
 
+
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+// TODO: Import postfixes
+@{event_type = '::'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]) + '_Event'}
+
+void *
+rosidl_@('_'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]))_introspection_message_create
+(
+    const rosidl_service_introspection_info_t * info,
+    rcutils_allocator_t * allocator,
+    void * request_message,
+    void * response_message,
+    bool enable_message_payload)
+{
+  auto * event_msg = static_cast<@event_type *>(allocator->zero_allocate(1, sizeof(@event_type), allocator->state));
+  if (nullptr == event_msg) {
+    return NULL;
+  }
+  event_msg = new(event_msg) @(event_type)(); 
+
+  event_msg->info.set__event_type(info->event_type);
+  event_msg->info.set__sequence_number(info->sequence_number);
+  event_msg->info.stamp.set__sec(info->stamp_sec);
+  event_msg->info.stamp.set__nanosec(info->stamp_nanosec);
+
+  std::array<uint8_t, 16> client_id;
+  std::move(std::begin(info->client_id), std::end(info->client_id), client_id.begin());
+  event_msg->info.client_id.set__uuid(client_id);
+
+  if (enable_message_payload) {
+    if (nullptr == request_message) {
+      event_msg->response.push_back(*static_cast<@('::'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]))_Response *> (response_message));
+    } else if (nullptr == response_message) {
+      event_msg->request.push_back(*static_cast<@('::'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]))_Request *> (request_message)); }
+      else {
+        // raise an error here?
+    }
+  }
+
+  return event_msg;
+}
+
+#include <memory>
+
+bool
+rosidl_@('_'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]))_introspection_message_destroy(
+void* event_msg,
+rcutils_allocator_t * allocator
+)
+{
+  auto * event_msg_ = static_cast<@event_type *>(event_msg);
+  // event_msg_->@('::'.join([package_name, *interface_path.parents[0].parts]) + '::~' + service.namespaced_type.name + '_Event')();
+  // std::destroy_at(event_msg_); // Why can't I compile with c++17?
```

Comment:
I guess if we want to rely on the c-allocators anyways, then we don't need to bother with `std::destroy_at` and can remove these commented lines.